src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -2498,6 +2499,337 @@ public static decimal Average<TSource>(this IEnumerable<TSource> source, Func<TS
         {
             return Enumerable.Average(Enumerable.Select(source, selector));
         }
+
+        public static TSource MaxBy<TSource, TKey>(this IEnumerable<TSource> source, Func<TSource, TKey> keySelector)
+        {
+            return source.MaxBy(keySelector, null);
+        }
```

Comment:
For perf it would probably make sense to check if source is T[] or IReadonlyList<T> and use a for loop.